catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -147,6 +157,48 @@ protected void onCreate(Bundle savedInstanceState) {
 			if (loadExternalProjectUri != null) {
 				loadProgramFromExternalSource(loadExternalProjectUri);
 			}
+
```

Comment:
for readability (and because of clean code guidelines), it would be nice to have the cast specific stuff in a function.